packages/account/src/Components/leave-confirm/leave-confirm.jsx

```diff
@@ -46,60 +46,55 @@ const LeaveConfirmMessage = ({ back, leave }) => {
  * Blocks routing if Formik form is dirty
  * Has to be a child of <Formik> for FormikConsumer to work
  */
-class TransitionBlocker extends React.Component {
-    state = { show: false };
+const TransitionBlocker = ({ dirty, onDirty }) => {
+    const [show, setShow] = React.useState(false);
+    const [next_location, setNextLocation] = React.useState(null);
+    const history = useHistory();
 
-    componentDidMount() {
-        this.unblock = this.props.history.block(next_location => {
-            if (this.props.dirty) {
-                if (this.props.onDirty) this.props.onDirty(false);
+    React.useEffect(() => {
+        return unblock;
```

Comment:
What is the intention of returning unblock here? Is it as a cleanup to call when the component is unmounted?

Reply:
it is to unblock the history on component unmount